File: src/gatling/scala/org/openapitools/client/model/ComponentsSchemasRecordCreateRequestItem.scala

package org.openapitools.client.model


case class ComponentsSchemasRecordCreateRequestItem (
    /* Form key identifying the eCRF to create or update */
    _formKey: String,
    /* Form ID identifying the eCRF to create or update (alternative to formKey) */
    _formId: Option[Integer],
    /* Name of the site where the record should be created (for new subject registration) */
    _siteName: Option[String],
    /* Site ID for the record (alternative to siteName) */
    _siteId: Option[Integer],
    /* Subject identifier (display ID) for which to create or update the record */
    _subjectKey: Option[String],
    /* Subject ID for which to create or update the record (alternative to subjectKey) */
    _subjectId: Option[Integer],
    /* Subject OID for which to create or update the record (alternative to subjectKey) */
    _subjectOid: Option[String],
    /* Name of the interval (visit) for a scheduled record update */
    _intervalName: Option[String],
    /* Interval ID for a scheduled record update (alternative to intervalName) */
    _intervalId: Option[Integer],
    /* Record ID for updating an existing unscheduled record (if applicable) */
    _recordId: Option[Integer],
    /* Record OID for updating an existing unscheduled record (if applicable) */
    _recordOid: Option[String],
    /* Key-value pairs of field names and values for the record data */
    _data: Map[String, AnyType]
)
object ComponentsSchemasRecordCreateRequestItem {
    def toStringBody(var_formKey: Object, var_formId: Object, var_siteName: Object, var_siteId: Object, var_subjectKey: Object, var_subjectId: Object, var_subjectOid: Object, var_intervalName: Object, var_intervalId: Object, var_recordId: Object, var_recordOid: Object, var_data: Object) =
        s"""
        | {
        | "formKey":$var_formKey,"formId":$var_formId,"siteName":$var_siteName,"siteId":$var_siteId,"subjectKey":$var_subjectKey,"subjectId":$var_subjectId,"subjectOid":$var_subjectOid,"intervalName":$var_intervalName,"intervalId":$var_intervalId,"recordId":$var_recordId,"recordOid":$var_recordOid,"data":$var_data
        | }
        """.stripMargin
}

File: client/zapier/models/components_schemas_RecordCreateRequest_item.js
const utils = require('../utils/utils');
const AnyType = require('../models/AnyType');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}formKey`,
                label: `Form key identifying the eCRF to create or update - [${labelPrefix}formKey]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}formId`,
                label: `Form ID identifying the eCRF to create or update (alternative to formKey) - [${labelPrefix}formId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}siteName`,
                label: `Name of the site where the record should be created (for new subject registration) - [${labelPrefix}siteName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}siteId`,
                label: `Site ID for the record (alternative to siteName) - [${labelPrefix}siteId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}subjectKey`,
                label: `Subject identifier (display ID) for which to create or update the record - [${labelPrefix}subjectKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}subjectId`,
                label: `Subject ID for which to create or update the record (alternative to subjectKey) - [${labelPrefix}subjectId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}subjectOid`,
                label: `Subject OID for which to create or update the record (alternative to subjectKey) - [${labelPrefix}subjectOid]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}intervalName`,
                label: `Name of the interval (visit) for a scheduled record update - [${labelPrefix}intervalName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}intervalId`,
                label: `Interval ID for a scheduled record update (alternative to intervalName) - [${labelPrefix}intervalId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}recordId`,
                label: `Record ID for updating an existing unscheduled record (if applicable) - [${labelPrefix}recordId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}recordOid`,
                label: `Record OID for updating an existing unscheduled record (if applicable) - [${labelPrefix}recordOid]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}data`,
                label: `Key-value pairs of field names and values for the record data - [${labelPrefix}data]`,
                required: true,
                dict: true,
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'formKey': bundle.inputData?.[`${keyPrefix}formKey`],
            'formId': bundle.inputData?.[`${keyPrefix}formId`],
            'siteName': bundle.inputData?.[`${keyPrefix}siteName`],
            'siteId': bundle.inputData?.[`${keyPrefix}siteId`],
            'subjectKey': bundle.inputData?.[`${keyPrefix}subjectKey`],
            'subjectId': bundle.inputData?.[`${keyPrefix}subjectId`],
            'subjectOid': bundle.inputData?.[`${keyPrefix}subjectOid`],
            'intervalName': bundle.inputData?.[`${keyPrefix}intervalName`],
            'intervalId': bundle.inputData?.[`${keyPrefix}intervalId`],
            'recordId': bundle.inputData?.[`${keyPrefix}recordId`],
            'recordOid': bundle.inputData?.[`${keyPrefix}recordOid`],
            'data': bundle.inputData?.[`${keyPrefix}data`],
        }
    },
}
